src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -0,0 +1,559 @@
+        
+        if error_list:
+            raise err_exit(fill('The following fields cannot be found: %r' % error_list))
+
+        payload = {"project_context":project, "fields":[{item:'$'.join(item.split('.'))} for item in fields_list]}
+        if "CohortBrowser" in resp['recordTypes']:
+            if resp.get('baseSql'):
+                payload['base_sql'] = resp.get('baseSql')
+            payload['filters'] = resp['filters']
+
+        if args.sql:
+            resource_val = resp['url'] + '/viz-query/3.0/' + resp['dataset'] + '/raw-query'
+            try:
+                resp_raw_query = dxpy.DXHTTPRequest(resource=resource_val, data=payload, prepend_srv=False)
+            except Exception as details:
+                raise ResolutionError(str(details))
+            sql_results = resp_raw_query['sql'] + ';'
+            if print_to_stdout:
+                print(sql_results)
+            else:
+                with open(out_file_field, 'w') as f:
+                    print(sql_results, file=f)
+        else:
+            resource_val = resp['url'] + '/data/3.0/' + resp['dataset'] + '/raw'
+            try:
+                resp_raw = dxpy.DXHTTPRequest(resource=resource_val, data=payload, prepend_srv=False)
+                if 'error' in resp_raw.keys():
+                    if resp_raw['error']['details'] and 'PermissionDenied: Access to the specified database is not allowed' in resp_raw['error']['details']:
+                        print("Insufficient permissions due to the project policy")
+                    else:
+                        print(resp_raw['error'])
+                    sys.exit(1)
+            except Exception as details:
+                raise ResolutionError(str(details))
+            csv_from_json(out_file_name=out_file_field, print_to_stdout=print_to_stdout, sep=delimiter, raw_results=resp_raw['results'], column_names=fields_list)
+
+    elif args.sql:
+        raise err_exit(fill('`--sql` passed without `--fields`'))
+        
+    
+    if args.dump_dataset_dictionary:
+        rec_dict = rec_descriptor.get_dictionary()
+        write_ot = rec_dict.write(output_file_data=output_file_data, output_file_entity=output_file_entity,
+                                  output_file_coding=output_file_coding, sep=delimiter)
+
+def csv_from_json(out_file_name="", print_to_stdout=False, sep=',', raw_results=[], column_names=[]):
+    if print_to_stdout:
+        fields_output = sys.stdout
+    else:
+        fields_output = open(out_file_name, 'w')
+
+    csv_writer = csv.DictWriter(fields_output, delimiter=sep, doublequote=True, escapechar = None, lineterminator = "\n", quotechar = str('"'), 
+                                quoting = csv.QUOTE_MINIMAL, skipinitialspace = False, strict = False, fieldnames=column_names)
+    csv_writer.writeheader()
+    for entry in raw_results:
+        csv_writer.writerow(entry)
+
+    if not print_to_stdout:
+        fields_output.close()
+    
+class DXDataset(DXRecord):
```

Comment:
Would it make sense to move this class definition into its separate file? To follow the python/dxpy/bindings example. But not necessary!:)

Reply:
Seeking opinion from other reviewers. Will let @kpjensen make the final call.

Reply:
In this PR I think it's fine to leave in this file as the classes are not used elsewhere within dxpy like the other DNAnexus object bindings.